punishments.js

```diff
@@ -54,6 +55,9 @@ class PunishmentMap extends Map {
 	has(/** @type {string} */ k) {
 		return !!this.get(k);
 	}
+	/**
+	 * @param {Function} callback
```

Comment:
You should really define a function signature here... Look at how I handle function signatures in LoginServer.

https://github.com/Zarel/Pokemon-Showdown/blob/da9cf7464b8ff278928e0208febbdc87897637c2/loginserver.js#L43

Reply:
Especially since if you define the function signature here, you won't have to define it every time you call `forEach`, like you're currently doing.